wandb/sdk/integration_utils/llm.py

```diff
@@ -91,11 +91,17 @@ def method(*args, **kwargs):
                     with Timer() as timer:
                         result = original_method(*args, **kwargs)
                     try:
-                        trace = self.resolver(
+                        metrics = self.resolver(
                             kwargs, result, timer.start_time, timer.elapsed
                         )
-                        if trace is not None:
-                            run.log({"trace": trace})
+                        if isinstance(metrics, trace_tree.WBTraceTree):
+                            run.log({"trace": metrics})
+                        elif isinstance(metrics, dict):
+                            run.log(metrics)
+                        else:
+                            logger.warning(
+                                f"Unable to log {self.name} metrics: {metrics}"
+                            )
```

Comment:
return dicts from all downstream resolvers so that there's no need for isinstance logic here.

```suggestion
                        if metrics:
                            run.log(metrics)
                        else:
                            logger.warning(
                                f"Unable to log {self.name} metrics: {metrics}"
                            )
```

Reply:
Perhaps a separate PR to change this ? I think @ash0ts is also working on a PR that uses this.